src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+    
+            $serviceBusAccount = Get-AzResource -ResourceGroupName $ResourceGroupName -Name $ServiceBusNamespace -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if(!$serviceBusAccount){
+                New-AzResource -ResourceGroupName $ResourceGroupName -Location $TargetRegion -Properties  $ServiceBusProperties -ResourceName $ServiceBusNamespace -ResourceType  $ServiceBusType -ApiVersion $ServiceBusApiVersion -Kind  $ServieBusKind -Sku  $ServiceBusSku -Tag $ResourceTag -Force
+                Write-Host $LogStringCreated, $ServiceBusNamespace
+            }else{
+                Write-Host $ServiceBusNamespace, $LogStringSkipping
+            }
+           
+            # Key vault
+            $KeyVaultName = $MigratePrefix + "kv" + $hash
+            $KeyVaultType = "Microsoft.KeyVault/vaults"
+            $KeyVaultApiVersion = "2016-10-01"
+            $KeyVaultKind = "KeyVault"
+            
+            $existingKeyVaultAccount =  Get-AzResource -ResourceGroupName $ResourceGroupName -Name $KeyVaultName -ErrorVariable notPresent -ErrorAction SilentlyContinue
+            if($existingKeyVaultAccount){
+                Write-Host $KeyVaultName, $LogStringSkipping
+            }else{
+                $tenantID = $context.Tenant.TenantId 
+
+                $KeyVaultPermissions = @{
+                    keys=@("Get","List","Create","Update","Delete");
+                    secrets=@("Get","Set","List","Delete");
+                    certificates=@("Get","List");
+                    storage= @("get","list","delete","set","update","regeneratekey","getsas",
+                    "listsas","deletesas","setsas","recover","backup","restore","purge")
+                }
+
+                $CloudEnvironMent = $context.Environment.Name
+                $HyperVManagerAppId = "b8340c3b-9267-498f-b21a-15d5547fd85e"
+                if($CloudEnvironMent -eq "AzureUSGovernment"){
+                    $HyperVManagerAppId = "AFAE2AF7-62E0-4AA4-8F66-B11F74F56326"
+                }
+                $hyperVManagerObject = Get-AzADServicePrincipal -ApplicationID $HyperVManagerAppId
+                $userObject =  Get-AzADUser -UserPrincipalName $context.Subscription.ExtendedProperties.Account
+                $accessPolicies = @()
+                $userAccessPolicy = @{
+                    "tenantId" = $tenantID;
+                    "objectId" =  $userObject.Id;
+                    "permissions" = $KeyVaultPermissions
+                }
+                $hyperVAccessPolicy = @{
+                    "tenantId" = $tenantID;
+                    "objectId" = $hyperVManagerObject.Id;
+                    "permissions" = $KeyVaultPermissions
+                }
+                $accessPolicies += $userAccessPolicy
+                $accessPolicies += $hyperVAccessPolicy
+                $projectRSPObject = Get-AzMigrateReplicationRecoveryServicesProvider -ResourceGroupName $ResourceGroupName -ResourceName $VaultName
```

Comment:
hope $projectRSPObject holds RSP that belongs to agentless fabric (of type VMwareV2). please cross check.

Reply:
Fixed it by if($projectRSP.FabricType -eq "VMwareV2"){